File: client/src/main/java/org/dreamexposure/discal/client/listeners/discord/ReadyEventListener.java
package org.dreamexposure.discal.client.listeners.discord;

import discord4j.core.event.domain.lifecycle.ReadyEvent;
import discord4j.rest.util.Image;
import org.dreamexposure.discal.client.message.Messages;
import org.dreamexposure.discal.core.utils.GlobalVal;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import reactor.core.publisher.Mono;

/**
 * @author NovaFox161
 * Date Created: 9/8/2018
 * For Project: DisCal-Discord-Bot
 * Author Website: https://www.novamaday.com
 * Company Website: https://www.dreamexposure.org
 * Contact: nova@dreamexposure.org
 */
@SuppressWarnings("OptionalGetWithoutIsPresent")
public class ReadyEventListener {
    private static final Logger LOGGER = LoggerFactory.getLogger(ReadyEventListener.class);

    public static Mono<Void> handle(final ReadyEvent event) {
        return event.getClient().getApplicationInfo()
            .doOnNext(info -> GlobalVal.setIconUrl(info.getIconUrl(Image.Format.PNG).get()))
            .doOnNext(info -> LOGGER.info(GlobalVal.getSTATUS(), "Ready event success!"))
            .then(Messages.reloadLangs())
            .onErrorResume(e -> {
                LOGGER.error(GlobalVal.getDEFAULT(), "Failed to handle ready event");
                return Mono.empty();
            })
            .then();

    }
}

File: client/src/main/kotlin/org/dreamexposure/discal/client/listeners/discord/ReadyEventListener.kt
package org.dreamexposure.discal.client.listeners.discord

import discord4j.core.event.domain.lifecycle.ReadyEvent
import discord4j.rest.util.Image
import kotlinx.coroutines.reactor.awaitSingle
import kotlinx.coroutines.reactor.awaitSingleOrNull
import org.dreamexposure.discal.client.message.Messages
import org.dreamexposure.discal.core.logger.LOGGER
import org.dreamexposure.discal.core.utils.GlobalVal.DEFAULT
import org.dreamexposure.discal.core.utils.GlobalVal.STATUS
import org.dreamexposure.discal.core.utils.GlobalVal.iconUrl
import org.springframework.stereotype.Component

@Component
class ReadyEventListener : EventListener<ReadyEvent> {
    override suspend fun handle(event: ReadyEvent) {
        try {
            iconUrl = event.client.applicationInfo
                .map { it.getIconUrl(Image.Format.PNG).orElse("") }
                .awaitSingle()

            Messages.reloadLangs().awaitSingleOrNull()

            LOGGER.info(STATUS, "Ready event success!")
        } catch (e: Exception) {
            LOGGER.error(DEFAULT, "Failed to handle ready event", e)
        }
    }
}
